fla/models/utils.py

```
@@ -2,149 +2,303 @@
+
+            if attn_state is not None:
+                input_size = attn_state[0].shape[1]
+                if self.state["attn_state"] is None:
+                    if window_size is not None and input_size > window_size:
+                        attn_state = tuple(x[:, -window_size:].contiguous() for x in attn_state)
+                    self.state["attn_state"] = tuple(attn_state)
+                else:
+                    old = self.state["attn_state"]
+                    if window_size is not None and old[0].shape[1] >= window_size:
+                        new_tuple = []
+                        for old_x, new_x in zip(old, attn_state):
```

Ruff (0.12.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)